src/Xhgui/Util.php

```diff
@@ -21,17 +24,102 @@ public static function simpleUrl($url)
     }
 
     /**
-     * Returns an new ObjectId-like string, where its first 8
-     * characters encode the current unix timestamp and the
-     * next 16 are random.
+     * Serialize data for storage
      *
-     * @see http://php.net/manual/en/mongodb-bson-objectid.construct.php
+     * @param      $data
+     * @param bool $profiles
+     *
+     * @return false|string
+     */
+    public static function getDataForStorage($data, $profiles = true)
+    {
+        if ($profiles) {
+            $serializer = Xhgui_Config::read('save.handler.serializer', 'json');
+        } else {
+            $serializer = Xhgui_Config::read('save.handler.meta_serializer', 'php');
+        }
+
+        switch ($serializer) {
+            case 'json':
+                return json_encode($data);
+                break;
+
+            case 'igbinary_serialize':
+            case 'igbinary_unserialize':
+            case 'igbinary':
+                return igbinary_serialize($data);
+                break;
+
+            case 'php':
+            case 'var_export':
+                return "<?php \n".var_export($data, true);
+                break;
+        }
+    }
+
+    /**
+     * Get id for a record.
+     *
+     * By default this method will try to re-use request id from http server.
+     * This is needed for some storage engines that don't have string/hashlike id generation.
+     *
+     * @param array $data
+     * @param bool $useRequestId
      *
      * @return string
      */
-    public static function generateId()
+    public static function getId(array $data = array(), $useRequestId = true)
     {
-        return dechex(time()) . bin2hex(random_bytes(8));
+
+        // in some cases, like during import, we might already have id
+        if (!empty($data['id'])) {
+            return $data['id'];
+        }
+
+        // mongo compatibility
+        if (!empty($data['_id'])) {
+            return $data['_id'];
+        }
+
+        if ($useRequestId) {
+            foreach(array('REQUEST_ID', 'HTTP_REQUEST_ID', 'HTTP_X_REQUEST_ID', 'X_CORRELATION_ID', 'HTTP_X_CORRELATION_ID') as $header) {
+                if (array_key_exists($header, $_SERVER) !== false) {
+                    return $_SERVER[$header];
+                }
+            }
+        }
+
+        // try php 7+ function.
+        if (function_exists('random_bytes')) {
```

Comment:
remove the `function_exists` check. and all the compatibility layer done below. the function is provided by php engine or the random-compat shim.

I wrote about this elsewhere in this PR.